utils/aeternity.js

```diff
@@ -97,26 +138,100 @@ class Aeternity {
       : fetchOracleState();
   };
 
+  static addAdditionalTipsData = (tips) => {
+    return tips.map(tip => {
+      tip.topics = [...new Set(tip.title.match(topicsRegex))].map((x) => x.toLowerCase());
+
+      tip.amount_ae = Util.atomsToAe(tip.amount).toFixed();
+      tip.total_amount_ae = Util.atomsToAe(tip.total_amount).toFixed();
+      tip.total_unclaimed_amount_ae = Util.atomsToAe(tip.total_unclaimed_amount).toFixed();
+      tip.total_claimed_amount_ae = Util.atomsToAe(tip.total_claimed_amount).toFixed();
+
+      tip.retips = tip.retips.map(retip => {
+        retip.amount_ae = Util.atomsToAe(retip.amount).toFixed();
+        return retip;
+      });
+
+      return tip;
+    })
+  }
+
   getTips = async () => {
     if (!this.client) throw new Error('Init sdk first');
     const fetchTips = async () => {
       const state = await this.contract.methods.get_state();
-      return this.getTipsRetips(state.decodedResult);
+      const tips = tippingContractUtil.getTipsRetips(state.decodedResult).tips;
+      return Aeternity.addAdditionalTipsData(tips)
     };
 
     return this.cache
       ? this.cache.getOrSet(['getTips'], () => fetchTips(), this.cache.shortCacheTime)
       : fetchTips();
   };
 
-  getContractSource = () => {
-    if (!process.env.CONTRACT_FILE) throw new Error(`env.CONTRACT_FILE is ${process.env.CONTRACT_FILE}`);
-    return fs.readFileSync(`${__dirname}/${process.env.CONTRACT_FILE}.aes`, 'utf-8');
-  };
+  getTokenRegistryState = async () => {
+    const fetchData = async () => {
+      const contract = await this.client.getContractInstance(TOKEN_REGISTRY, {contractAddress: process.env.TOKEN_REGISTRY_ADDRESS});
+      return contract.methods.get_state().then(r => r.decodedResult);
+    }
 
-  getOracleContractSource = () => {
-    return fs.readFileSync(`${__dirname}/OracleServiceInterface.aes`, 'utf-8');
-  };
+    return this.cache
+      ? this.cache.getOrSet(["getTokenRegistryState"], () => fetchData(), this.cache.shortCacheTime)
+      : fetchData();
+  }
+
+  getTokenMetaInfoCacheAccounts = async (address) => {
+    const fetchData = async () => {
+      const contract = await this.client.getContractInstance(TOKEN_CONTRACT_INTERFACE, {contractAddress: address});
```

Comment:
do we need to re-instance the contract everytime someone looks for their balance?

Reply:
`getTokenMetaInfoCacheAccounts` itself is cached, caching the instances or keeping them is an option, but I'm not sure its worth it, I'll check